client/my-sites/controller.js

```diff
@@ -457,7 +457,7 @@ export function siteSelection( context, next ) {
 	}
 
 	// If the path fragment does not resemble a site, set all sites to visible
-	if ( ! siteFragment ) {
+	if ( ! ( typeof siteFragment === 'string' && siteFragment.length ) ) {
```

Comment:
A few lines below on line 481, `siteFragment.replace` errors out with `o.replace is not a function`. This confused me for a moment, because we should _never_ reach that line of code if the siteFragment is falsey. The only possibility I can think of is that there's a type mismatch of some sort, or that the length check isn't working out. 

Unless some other part of the logic is wonky, this should harden the code a bit to prevent the error.